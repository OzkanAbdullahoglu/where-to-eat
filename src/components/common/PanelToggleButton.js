import styled from 'styled-components';

const backgroundURL =
  'url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAUCAQAAAAXDMSnAAAAi0lEQVR4AX3JQcqBURQG4O/' +
  '+9WNG30D3vOfSDTuQsgcZyBakZANSzMVMBme3zsBI5/VMn4ZKLP5ki1E4tYejWpilxVUtzOEUD68odYmXR5BJNp/' +
  '4zllXD2phllYvamHmirsayUkfJ5ruHzueTldC08kcT5YOY9xYujqQM03XKXuaLmEtNF1e1Nz89gbL+0do6OEwRwAAAABJRU5ErkJggg==)';

export const PanelToggleButton = styled.button`
         width: 23px;
         height: 48px;
         cursor: pointer;
         background: rgba(255, 255, 255, 0.9) ${backgroundURL} 7px center/7px
           10px no-repeat;
         border-left: 1px solid #d4d4d4;
         box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.3);
         -webkit-transform: ${(props) =>
    props.closed === true ? 'scaleX(-1)' : 'scaleX(1)'};
         transform: ${(props) =>
    props.closed === true ? 'scaleX(-1)' : 'scaleX(1)'};
         @media (max-width: 414px) {
          display: none;
         }
       `;

