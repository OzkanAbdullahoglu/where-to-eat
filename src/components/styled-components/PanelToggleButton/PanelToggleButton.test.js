/* eslint-disable linebreak-style */
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import PanelToggleButton from './PanelToggleButton';

describe('Panel Toggle Button Render', () => {
  const props = {
    closed: true,
    hide: false,
  };

  it('renders toggle button', () => {
    const toggleButtonWrapper = shallow(<PanelToggleButton {...props} />);
    expect(toJson(toggleButtonWrapper)).toMatchSnapshot();
  });
  it('renders `closed` and `hide`  props', () => {
    const toggleButtonWrapper = shallow(<PanelToggleButton {...props} />);
    expect(toggleButtonWrapper).toHaveStyleRule('transform', 'scaleX(-1)');
    expect(toggleButtonWrapper).toHaveStyleRule('display', 'inline-block');
  });
  it('renders properly on modifiers and specific media queries', () => {
    const toggleButtonWrapper = shallow(<PanelToggleButton {...props} />);
    expect(toggleButtonWrapper).toHaveStyleRule('background-color', '#fffafb', {
      modifier: ':hover',
    });
    expect(toggleButtonWrapper).toHaveStyleRule('display', 'none', {
      media: '(max-width:414px)',
    });
  });
});
