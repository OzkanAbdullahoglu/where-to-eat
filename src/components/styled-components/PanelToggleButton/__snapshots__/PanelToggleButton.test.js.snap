// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Toggle Button Render renders toggle button 1`] = `
.c0 {
  width: 23px;
  height: 48px;
  cursor: pointer;
  background: rgba(255,255,255,0.9) url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAUCAQAAAAXDMSnAAAAi0lEQVR4AX3JQcqBURQG4O/+9WNG30D3vOfSDTuQsgcZyBakZANSzMVMBme3zsBI5/VMn4ZKLP5ki1E4tYejWpilxVUtzOEUD68odYmXR5BJNp/4zllXD2phllYvamHmirsayUkfJ5ruHzueTldC08kcT5YOY9xYujqQM03XKXuaLmEtNF1e1Nz89gbL+0do6OEwRwAAAABJRU5ErkJggg==) 7px center/7px 10px no-repeat;
  border-left: 1px solid #d4d4d4;
  box-shadow: 0px 1px 4px rgba(0,0,0,0.3);
  -webkit-transform: scaleX(-1);
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
  display: inline-block;
}

.c0:hover {
  background-color: #fffafb;
}

@media (max-width:414px) {
  .c0 {
    display: none;
  }
}

<button
  className="c0"
/>
`;
